RecoLocalTracker/SiPixelClusterizer/plugins/gpuClustering.h

```diff
@@ -79,9 +81,11 @@ namespace gpuClustering {
     if (first>=msize) return;
 
     int jmax[10];
+    // int jmin[10];
     auto niter = (msize-first)/blockDim.x;
     assert(niter<10);
-    for (int i=0; i<niter+1; ++i) jmax[i]=msize;    
+    // for (int k=0; k<niter+1; ++k) jmin[k]=first+k*blockDim.x+1;    
+    for (int k=0; k<niter+1; ++k) jmax[k]=msize;
 
     while (go) {
```

Comment:
I know it was like this before, but I'm wondering - is it actually safe to use `__syncthreads()` inside a `while` loop?
Isn't it a problem if some threads run a different number of loops, or take different branches (i.e. `continue`) ?

Reply:
cuda-memcheck  did not detect any race

Reply:
All threads goes through the while loop together.
the continue is in the for and there is no ```__syncthreads()``` in the for-loop.
So no divergencies besides the for loop